src/Core/OntologyAnnotation.fs

```diff
@@ -131,31 +131,28 @@ type OntologyAnnotation(?name,?tsr,?tan, ?comments) =
         sb.Append("}") |> ignore
         sb.ToString()
 
-    override this.Equals other =
-        match other with
-        | :? OntologyAnnotation as oa -> (this :> System.IEquatable<_>).Equals oa
-        | :? string as s ->           
-            this.NameText = s
-            || 
-            this.TermAccessionShort = s
-            ||
-            this.TermAccessionOntobeeUrl = s
-        | _ -> false
-
-    override this.GetHashCode () = (this.NameText+this.TermAccessionShort).GetHashCode()
-
-    interface System.IEquatable<OntologyAnnotation> with
-        member this.Equals other =
-            if this.TermAccessionNumber.IsSome && other.TermAccessionNumber.IsSome then
-                other.TermAccessionShort = this.TermAccessionShort
-                ||
-                other.TermAccessionOntobeeUrl = this.TermAccessionOntobeeUrl
-            elif this.Name.IsSome && other.Name.IsSome then
-                other.NameText = this.NameText
-            elif this.TermAccessionNumber.IsNone && other.TermAccessionNumber.IsNone && this.Name.IsNone && other.Name.IsNone then
-                true
-            else 
-                false
+    override this.GetHashCode() = 
+        [|
+            HashCodes.boxHashOption this.Name
+            match this.TermSourceREF, this.TANInfo with
+            | None, Some taninfo -> // if we get taninfo we assume tsr to be inferrable by taninfo
+                HashCodes.hash {|tsr = taninfo.IDSpace; tan = taninfo.IDSpace + ":" + taninfo.LocalID|}
```

Comment:
I would make this explicit, i.e. call hashes on the fields and merge them. Record Type hashes behave a bit weird for fable Python

Reply:
Yeah really not sure about this. I tried to remove the python specific code and it returned strange numbers. But i must check tomorrow in more detail